lib/Doctrine/MongoDB/Aggregation/Stage/AbstractBucket.php

```diff
@@ -57,6 +57,21 @@ public function getExpression()
     }
 
     /**
+     * Converts an expression object into an array, recursing into nested items
+     *
+     * For expression objects, it calls getExpression on the expression object.
+     * For arrays, it recursively calls itself for each array item. Other values
+     * are returned directly.
+     *
+     * @param mixed|self $expression
+     * @return string|array
+     */
+    protected function convertExpression($expression)
```

Comment:
Introducing this method is necessary to allow mongodb-odm to override it and convert the expression using classMetadata from the document the aggregation builder is using. This would require us to release this bugfix as 1.6.0 since protected methods in non-final classes are part of the public API. However, I couldn't find a good way to solve this issue without introducing another method. What do you think @jmikola @malarzm?